core/serialization/serialization.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Contains top-level functions for serialization of the workspace
+ *     to JavaScript objects.
+ */
+'use strict';
+
+/**
+ * The top level namespace for JavaScript Object serialization.
+ * @namespace Blockly.serialization
+ */
+goog.module('Blockly.serialization');
+goog.module.declareLegacyNamespace();
+ 
+const Events = goog.require('Blockly.Events');
+const PluginSerializer = goog.require('Blockly.serialization.PluginSerializer');
+const blocks = goog.require('Blockly.serialization.blocks');
+const dom = goog.require('Blockly.utils.dom');
+const exceptions = goog.require('Blockly.serialization.exceptions');
+const registry = goog.require('Blockly.registry');
+const variables = goog.require('Blockly.serialization.variables');
+ 
+
+/**
+ * Returns the state of the workspace as a plain JavaScript object.
+ * @param {!Blockly.Workspace} workspace The workspace to serialize.
+ * @return {!Object<string, *>} The serialized state of the workspace.
+ */
+const save = function(workspace) {
+  const state = Object.create(null);
+
+  const serializerMap = registry.getAllItems(registry.Type.PLUGIN_SERIALIZER);
```

Comment:
It may be a good idea to add some wrappers instead of using the registry directly. Eg `serialization.workspaces.registerSerializer`. That way if we decide the registry doesn't make sense for plugin serializers we can easily change it.

Reply:
Sorry I didn't see this earlier. Yes, I think that is a great idea.